packages/bitcore-wallet-client/src/lib/api.ts

```diff
@@ -2076,26 +2076,36 @@ export class API extends EventEmitter {
           const serializedTxs =
             typeof serializedTx === 'string' ? [serializedTx] : serializedTx;
 
-          let i = 0;
+
+          const weightedSize = [];
+
 
           let isBtcSegwit =
             txp.coin == 'btc' &&
             (txp.addressType == 'P2WSH' || txp.addressType == 'P2WPKH');
+
+          let i = 0;
           for (const unsigned of unserializedTxs) {
-            let size = serializedTxs[i++].length / 2;
+            let size;
             if (isBtcSegwit) {
-              let unsignedSize = unsigned.length / 2;
-              size = Math.floor(size - (unsignedSize * 3) / 4);
+              // we dont have a fast way to calculate weigthedSize`
+              size = Math.floor(txp.fee/txp.feePerKb*1000) - 10;
```

Comment:
because fee is calculated with good precision on bitcore/bws, this should be fine.

Reply:
So, `weightedSize` would have same value in each elements in the array, that's correct?

Reply:
yes! on BTC/BCH paypro has only 1 instruction, there is only 1 rawTx.